bazel/envoy_build_system.bzl

```diff
@@ -78,6 +78,7 @@ def envoy_cmake_external(
         pdb_name = "",
         cmake_files_dir = "$BUILD_TMPDIR/CMakeFiles",
         generate_crosstool_file = False,
+        variants = [],
```

Comment:
I don't really think you need `variants`? Can we just do `select` on `lib_source` and `cache_entries`? Then it won't affect curl at all?

Reply:
Sorry, a basic question, 

I think I tried to have something like following

```
    cache_entries = select({
        "//conditions:default": {
            "BUILD_SHARED_LIBS": "off",
            "CMAKE_CXX_COMPILER_FORCED": "on",
            "CMAKE_C_COMPILER_FORCED": "on",
            "SKIP_BUILD_EXAMPLES": "on",
        },
        "//bazel:zlib_ng": {
             ...

```


How to get around the following error?

```
external/envoy/bazel/envoy_build_system.bzl", line 82, in envoy_cmake_external
                cache_entries.update
```

Seems like `select()` is not evaluated before we pass the value to `cache_entries`?

Reply:
yeah, you need to refactor that part in the envoy_build_system implementation, basically make it select aware.

Reply:
I think I have resolved this. Thanks for the tip!